community/kernel/src/docs/ops/performance-guide.asciidoc

```diff
@@ -1,28 +1,33 @@
 
-== Try this first ==
 
-The first thing to look at, when Neo4j is not performing as expected, is to make sure that your Cypher queries do not do more work than they have to.
-For instance, a query might, unbeknownst to the author, mandate the production of a large cartesian product; or it might perform an expensive label-scan, because a certain label/property combination isn't indexed.
+== Try this first
+
+The first thing to look at when Neo4j is not performing as expected is how the Cypher queries are being executed.
+Make sure that they don't do more work than they have to.
+Some queries may accidentally be written in a way that generates a large cartesian product.
+Other queries may have to perform expensive label scans because a label is missing an important index.
 The <<query-tuning>> chapter has more information on how to investigate Cypher performance issues.
 
-The second thing to look at, is to make sure that the Neo4j Java process has enough memory to do its work.
-If there is not enough memory to keep the JVM heap resident, then the OS will swap it out to storage.
-When a garbage collection happens, the swapped out heap memory has to be swapped in again, and something else will have to be swapped out.
-This swap-thrashing effect has a dramatic impact on the performance of the database, rendering it practically unusable.
+The second thing to look at is the Java Virtual Machine process.
+Make sure that it has enough memory to do its work.
+If there is not enough memory to keep the JVM heap resident, the operating system will swap it out to storage.
+When a garbage collection happens, the swapped out heap memory has to be swapped back in, and something else swapped out.
+This is called _swap-thrashing_ and has dramatic impact on the performance of a database, rendering it practically unusable.
 A well-tuned Neo4j database should not have any swap activity in its steady-state.
 
-Next, make sure the JVM has enough memory, and isn't spending too much time in garbage collection.
-The goal is to have a large enough heap so heavy/peak load will not result in so called GC-trashing.
+Next, make sure that the JVM has enough memory, and isn't spending too much time in garbage collection.
+The goal is to have a large enough heap to handle peak load without thrashing occurring in the garbage collector.
 Performance can drop as much as two orders of a magnitude when GC-thrashing happens.
 
-Start the JVM with `-server` flag and `-Xmx<good sized heap>`, f.ex. `-Xmx512m` for 512 MiB memory or `-Xmx3g` for 3GiB memory.
-Having too large heap may also hurt performance so you may have to try out some different heap sizes.
-Make sure you are using a concurrent garbage collector.
-We find that `-XX:+UseG1GC` works well in most use-cases.
+Start the JVM with `-server` flag and `-Xmx<good sized heap>`, for example `-Xmx512m` for 512 MB memory or `-Xmx3g` for 3 GB memory.
```

Comment:
This is not intended to change the meaning, but to standardize notation. Elsewhere in the manual we use (inconsistently cased and spaced) `MB` for the binary unit. This may be inappropriate in a scientific context, but in a user manual I believe it is appropriate, despite the ambiguity. I don't believe we use `MiB` anywhere else in the manual. You are the second person to remark on this, so I want to revisit the question–it's possible we should standardize on `MiB` et alia. That would be a separate decision, however, out of scope for this PR.